controllers/serviceclaim_controller.go

```diff
@@ -301,6 +302,15 @@ func (r *ServiceClaimReconciler) processServiceClaim(
 	}
 
 	if !registeredServiceFound {
+		c := metav1.Condition{
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Type:               primazaiov1alpha1.ServiceBindingConditionReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "UnmatchedSCI",
+			Message:            "",
```

Comment:
I'd propose to use a less technical reason 

```suggestion
			Reason:             "NoMatchingServiceFound",
			Message:            err.Error(),
```

Reply:
The error object is not available. But I have updated the message. Please check.